homeassistant/components/upnp/__init__.py

```diff
@@ -106,24 +109,30 @@ async def device_discovered(
     device.original_udn = entry.data[CONFIG_ENTRY_ORIGINAL_UDN]
 
     # Store mac address for changed UDN matching.
-    if device.host:
-        device.mac_address = await async_get_mac_address_from_host(hass, device.host)
-    if device.mac_address and not entry.data.get("CONFIG_ENTRY_MAC_ADDRESS"):
+    device_mac_address = await device.async_get_mac_address()
+    if device_mac_address and not entry.data.get("CONFIG_ENTRY_MAC_ADDRESS"):
```

Comment:
Is this meant to be quoted?

There is also a `CONFIG_ENTRY_MAC_ADDRESS` constant

Reply:
Good catch!